addon/components/connect.js

```diff
@@ -38,14 +39,31 @@ var connect = function(mapStateToComputed, mapDispatchToActions) {
                     component['actions'][action] = finalMapDispatchToActions(redux.dispatch.bind(redux))[action];
                 });
                 if (shouldSubscribe && !this.unsubscribe) {
-                    this.unsubscribe = redux.subscribe(() => {
-                        props.forEach(function(name) {
-                            component.notifyPropertyChange(name);
-                        });
-                    });
+                    this.unsubscribe = redux.subscribe(this.handleChange.bind(this));
                 }
                 this._super(...arguments);
             },
+            handleChange() {
+                var redux = this.get('redux');
+                var props = mapState(redux.getState());
+                var componentState = this.getComponentState(props);
+                var reduxState = finalMapStateToComputed(redux.getState());
+                if (!shallowEqual(componentState, reduxState)) {
```

Comment:
<del>This is a great implementation and thanks for keeping up the good work on this, but is there anyway you can do something like:</del>
```javascript
handeChange() { 
  var redux = ...
  ...
  const isEqual = this.getWithDefault('stateEqual', shallowEqual);
  ...
}
```
<del>And thereby expose an api for customizing our equality checker. This will be particularly useful for a lot of us who are using [ember-immutable](https://www.npmjs.com/package/ember-immutable), [seamless-immutable](https://www.npmjs.com/package/ember-seamless-immutable), or maybe even some custom library that ships with its own equality checker.</del>

edit: never-mind, I'm stupid, lol. You're already putting state into flat pojos